flytekit/image_spec/image_spec.py

```diff
@@ -95,6 +96,7 @@ class ImageSpec:
     source_copy_mode: Optional[CopyFileDetection] = None
     copy: Optional[List[str]] = None
     python_exec: Optional[str] = None
+    builder_config: Optional[typing.Dict[str, str]] = None
```

Comment:
Can the typing be `Optional[typing.Dict[str, typing.Any]]` or just `Optional[dict]`? 

We may extend more features where the value is not a str.